components/script/dom/readablestreamdefaultcontroller.rs

```diff
@@ -306,24 +324,71 @@ impl ReadableStreamDefaultController {
         // perform ! ReadableStreamFulfillReadRequest(stream, chunk, false).
         if stream.is_locked() && stream.get_num_read_requests() > 0 {
             stream.fulfill_read_request(chunk, false);
-            return;
+            return Ok(());
         }
 
-        // Otherwise, perform EnqueueValueWithSize(controller, chunk, chunkSize).
+        // Let result be the result of performing controller.[[strategySizeAlgorithm]],
+        // passing in chunk, and interpreting the result as a completion record.
+        let size = if let Some(strategy_size) = self.strategy_size.borrow().as_ref() {
+            let result = strategy_size.Call__(chunk, ExceptionHandling::Report);
+            match result {
+                // Let chunkSize be result.[[Value]].
+                Ok(size) => size,
+                Err(error) => {
+                    // If result is an abrupt completion,
+                    rooted!(in(*cx) let mut rval = UndefinedValue());
+                    
+                    // TODO: check if this is the right globalscope.
+                    unsafe {
+                        error
+                            .clone()
+                            .to_jsval(*cx, &*self.global(), rval.handle_mut())
+                    };
+                    
+                    // Perform ! ReadableStreamDefaultControllerError(controller, result.[[Value]]).
+                    self.error(rval.handle());
+                    
+                    // Return result.
+                    return Err(error);
```

Comment:
Looks like there is some duplication again in the error handling, but I suggest for do these clean-ups later(who knows what other stuff will emerge as we implement the rest).

Reply:
Indeed let do clean-ups later.